docs/curl/datatx.md

```diff
@@ -1,213 +1,412 @@
+```
+./bin/signer -privatekey=deployment/sample/crypto/alice/alice.key -data='{"must_sign_user_ids":["alice"],"tx_id":"1b6d6414-9b58-45d0-9723-1f31712add81","db_operations":[{"db_name":"db2","data_writes":[{"key":"key1","value":"eXl5","acl":{"read_users":{"alice":true,"bob":true},"read_write_users":{"alice":true}}}]}]}'
+```
+**Output**
+```
+MEQCIAM/FYzdfVlQGWBPcyDMp2BRDyzQdTdusOl0M/UBCk2gAiAxns+4m30Y/HzlO0e0dK0HnaWhbxch5tUys0P0ME7ZPw==
+```
+
+Once the above transaction gets committed, the submitter of the transaction would receive the following transaction receipt
+```json
+{
+  "response": {
+    "header": {
+      "node_id": "bdb-node-1"
+    },
+    "receipt": {
+      "header": {
+        "base_header": {
+          "number": 5,
+          "previous_base_header_hash": "0hi2UZF5+X4xcGhobE1akVFZ/depxYCWShukzF+KrNk=",
+          "last_committed_block_hash": "t9F0hOUPIuyk8upeQhpDpTY/wLP3yoLuA7NBqNE0HUo=",
+          "last_committed_block_num": 4
+        },
+        "skipchain_hashes": [
+          "t9F0hOUPIuyk8upeQhpDpTY/wLP3yoLuA7NBqNE0HUo=",
+          "PAFz7foMQVLrK0q1bO9TYxW9ZwUMl7rb6G3DZoSD1FY=",
+          "xz3tsHvNxVNsB709SBOJ3s2fuUXo2GciBGThc70aaTM="
+        ],
+        "tx_merkel_tree_root_hash": "UTf7CV8x9/QC9luiVwe5QzHVg9hvCHNLxWAC2iD/5Zw=",
+        "state_merkel_tree_root_hash": "+kBbsknNLNuDVKxDTktcgcJMsR9gRAHHW03plIjrTg0=",
+        "validation_info": [
+          {}
+        ]
+      }
+    }
+  },
+  "signature": "MEQCIDwJOH8ZNqs2H1rD+GWTTm03Cu1X+LcILP5NnPpmZ14LAiAJHmD4/E2noofY47eyjJm5NCcAmTUytqoB8Dat3xeVaQ=="
+}
+```
+
+### Checking the existance of the state
 
-Once the above transaction gets committed, we can query `key1` as follows:
+Let's query the node to see whether `key1` exists. The query can be submitted by either `alice` or `bob` as both have
+the read permission to this key. No one else can read `key1` including the admin user of the node. In this example,
+we use `bob` to query the key.
 
+First, compute the bob signature on the request payload.
+```sh
+./bin/signer -privatekey=deployment/sample/crypto/bob/bob.key -data='{"user_id":"bob","db_name":"db2","key":"key1"}'
+```
+**Output**
+```
+MEUCIQDm6dLmAdd0X49JygTiUkh+brZxprWSr2+hcAH+QIu3AAIgF+m7kO33YXyyqSbnXS9HR79wt/aL3JGhKvXFQaFBJms=
+```
+Second, submit the query
 ```sh
 curl \
      -H "Content-Type: application/json" \
-     -H "UserID: user1" \
-     -H "Signature: abcd" \
-     -X GET http://127.0.0.1:6001/data/bdb/key1 | jq .
+     -H "UserID: bob" \
+     -H "Signature: MEUCIQDm6dLmAdd0X49JygTiUkh+brZxprWSr2+hcAH+QIu3AAIgF+m7kO33YXyyqSbnXS9HR79wt/aL3JGhKvXFQaFBJms=" \
+     -X GET http://127.0.0.1:6001/data/db2/key1 | jq .
 ```
+
 **Output**
 ```json
 {
-  "payload": {
+  "response": {
     "header": {
-      "nodeID": "bdb-node-1"
+      "node_id": "bdb-node-1"
     },
-    "value": "dmFsdWUy",
+    "value": "eXl5",
     "metadata": {
       "version": {
-        "block_num": 28
+        "block_num": 5
       },
```

Comment:
It is confusing that the version does not include the index.... I guess it is committed because it is 0

Reply:
Yes. Due to `omitempty`, we have this issue. We should remove `omitempty` from all proto messages. Otherwise, even the read of existing data/users looks incomplete. I have created an issue https://github.com/IBM-Blockchain/bcdb-server/issues/109 to address this.